spec/safe_statements_spec.rb

```diff
@@ -1263,6 +1281,239 @@ def up
           end
         end
 
+        describe "safe_add_concurrent_partitioned_index" do
+          before do
+            ActiveRecord::Base.connection.execute(<<~SQL)
+              CREATE SCHEMA partman;
+              CREATE EXTENSION pg_partman SCHEMA partman;
+            SQL
+          end
+
+          it "creates valid index when there are no child partitions" do
+            create_range_partitioned_table(:foos3, migration_klass)
+
+            test_migration = Class.new(migration_klass) do
+              def up
+                safe_add_concurrent_partitioned_index :foos3, :updated_at
+              end
+            end
+
+            allow(ActiveRecord::Base.connection).to receive(:execute).and_call_original
+
+            test_migration.suppress_messages { test_migration.migrate(:up) }
+
+            expect(ActiveRecord::Base.connection).to have_received(:execute).with(/CREATE INDEX/).once
+            expect(ActiveRecord::Base.connection).to_not have_received(:execute).with(/ALTER INDEX/)
+
+            indexes = ActiveRecord::Base.connection.indexes(:foos3)
+
+            expect(indexes.size).to eq(1)
+            expect(indexes.first).to have_attributes(
+              table: :foos3,
+              name: "foos3_updated_at_idx",
+              columns: ["updated_at"],
+            )
+          end
+
+          it "creates valid index when multiple child partitions exist" do
+            create_range_partitioned_table(:foos3, migration_klass, with_partman: true)
+
+            test_migration = Class.new(migration_klass) do
+              def up
+                safe_add_concurrent_partitioned_index :foos3, :updated_at
+              end
+            end
+
+            allow(ActiveRecord::Base.connection).to receive(:execute).and_call_original
+
+            test_migration.suppress_messages { test_migration.migrate(:up) }
+
+            expect(ActiveRecord::Base.connection).to have_received(:execute).with(/CREATE INDEX/).exactly(11).times
+            expect(ActiveRecord::Base.connection).to have_received(:execute).with(/ALTER INDEX/).exactly(10).times
+
+            # look up indexes on child tables and parent table
+            partitions_for_table(:foos3).append(:foos3).each do |table|
```

Comment:
Can we wrap this in `aggregate_failures`?

Reply:
TIL!